packages/composable-controller/src/ComposableController.ts

```diff
@@ -157,27 +160,29 @@ export class ComposableController<
 
     super({
       name: controllerName,
-      metadata: controllers.reduce<StateMetadata<ComposableControllerState>>(
-        (metadata, controller) => ({
-          ...metadata,
-          [controller.name]: isBaseController(controller)
-            ? controller.metadata
-            : { persist: true, anonymous: true },
-        }),
-        {} as never,
-      ),
-      state: controllers.reduce<ComposableControllerState>(
+      metadata: Object.keys(controllers).reduce<
+        StateMetadata<ComposableControllerState>
+      >((metadata, name) => {
+        (metadata as StateMetadataConstraint)[name] = {
+          persist: true,
+          anonymous: true,
+        };
+        return metadata;
```

Comment:
Can we avoid the type assertion by doing the following?

``` suggestion
        return {
          ...metadata,
          [name]: {
            persist: true,
            anonymous: true,
          },
        };
```

Reply:
See https://github.com/MetaMask/core/pull/4968#discussion_r1868166507